app/registries/foreman/plugin.rb

```diff
@@ -317,6 +317,24 @@ def tests_to_skip(hash)
       end
     end
 
+    # Adds setting definition
+    #
+    # ===== Example
+    #
+    #   settings do
+    #     category(:cfgmgmt, N_('Configuration Management')) do
+    #       setting(:use_cooler_puppet, default: true,
+    #                                   description: N_('This will ensure we install a new and shiny Puppet on hosts instead of a rusty one'),
+    #                                   full_name: N_('Use shiny puppet'),
+    #                                   encrypt: true)
+    #       end
+    #     end
+    #   end
+    #
+    def settings(&block)
+      SettingManager.map(id, &block)
+    end
```

Comment:
Nitpick: I hate staring lines at column no 50 or above, if you would do C-like ANSI formatting (newline after cooler_puppet) that could be actually nicer but it is just my taste.